### PR TITLE
build: bump rxjs to 7.5.5

### PR DESCRIPTION
**Pull request by haroldwaters**
Add in types to allow version bump to RXJS 7

### CODE REVIEW

**Review by AndrewBird81 — APPROVED** [review]

File: lib/server/server-google-pubsub.ts [hunk 1]
```diff
@@ -251,9 +251,9 @@ export class GooglePubSubTransport extends Server implements CustomTransportStra
         const _topicName = this.topicNamingStrategy.generateTopicName(topicName);
         const topic: GooglePubSubTopic | null = this.googlePubSubClient.getTopic(_topicName);
 
-        return await this.googlePubSubClient
+        return (await this.googlePubSubClient
             .createSubscription(subscriptionName, topic, createOptions)
-            .toPromise();
+            .toPromise()) as GooglePubSubSubscription | null;
```
Comment (jordangarrison) [inline lib/server/server-google-pubsub.ts:256]: What's the reason for the bulk option here?

**Review by jordangarrison — APPROVED** [review]
This looks good, just one question below but going ahead and approving. What will the upgrade process look like? Do we need to do a full NestJS upgrade?

File: lib/server/server-google-pubsub.ts (hunk 1 above)
Comment (haroldwaters) [reply lib/server/server-google-pubsub.ts:256]: Convention with in the library.

**Comment by haroldwaters** [comment]
> This looks good, just one question below but going ahead and approving. What will the upgrade process look like? Do we need to do a full NestJS upgrade?

The full upgrade is going to involve removing `toPromise()` throughout - though that can wait until the next major RXJS version.

**Pull request merged**